contracts/actions/morpho-blue/MorphoBlueSupplyCollateral.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import "../ActionBase.sol";
+import "../../utils/TokenUtils.sol";
+import "./helpers/MorphoBlueHelper.sol";
+
+/// @title Supply a token to Morpho Blue market as collateral
+contract MorphoBlueSupplyCollateral is ActionBase, MorphoBlueHelper {
+    using TokenUtils for address;
+
+    struct Params {
+        MarketParams marketParams;
+        uint256 supplyAmount;
+        address from;
+        address onBehalf;
+    }
+
+    function executeAction(
+        bytes memory _callData,
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+
+        params.marketParams.loanToken = _parseParamAddr(params.marketParams.loanToken , _paramMapping[0], _subData, _returnValues);
+        params.marketParams.collateralToken = _parseParamAddr(params.marketParams.collateralToken , _paramMapping[1], _subData, _returnValues);
+        params.marketParams.oracle = _parseParamAddr(params.marketParams.oracle , _paramMapping[2], _subData, _returnValues);
+        params.marketParams.irm = _parseParamAddr(params.marketParams.irm , _paramMapping[3], _subData, _returnValues);
+        params.marketParams.lltv = _parseParamUint(params.marketParams.lltv, _paramMapping[4], _subData, _returnValues);
```

Comment:
Could this be factorized to `MorphoBlueHelper`? To avoid redundancy thus be less prone to error in the order it is loaded

Reply:
Not sure i follow completely, but this is the way it needs to be due to our Recipe/Strategy systems (we'd most likely parse these 5 from users subscription data inside a strategy that's onchain)

Reply:
I mean the order and types of market params are pre-defined and common to all interactions that involve market params on Blue, so the decoding of these could be factorized to an internal function, for example defined in MorphoBlueHelper, so you avoid making errors in the order and types of market params when decoding these in each interaction